File: rb101/rb101-rb109-small_problems/easy8/fizzbuzz.rb
def fizzbuzz_value(num)
  case
  when (num % 3).zero? && (num % 5).zero? then 'FizzBuzz'
  when (num % 3).zero? then 'Fizz'
  when (num % 5).zero? then 'Buzz'
  else num
  end
end

def fizzbuzz(num1, num2)
  arr = []
  (num1..num2).each do |num|
    arr << fizzbuzz_value(num)
  end
  puts arr.join(', ')
end

fizzbuzz(1, 15)

# ALTERNATIVE SOLUTION:

# def fizzbuzz(starting_number, ending_number)
#   result = []
#   starting_number.upto(ending_number) do |number|
#     result << fizzbuzz_value(number)
#   end
#   puts result.join(', ')
# end

# def fizzbuzz_value(number)
#   case
#   when number % 3 == 0 && number % 5 == 0
#     'FizzBuzz'
#   when number % 5 == 0
#     'Buzz'
#   when number % 3 == 0
#     'Fizz'
#   else
#     number
#   end
# end

File: js210/lesson_3/fizzBuzz.js
function fizzBuzz() {
  for (let number = 1; number <= 100; number++) {
    let message = number;

    if (number % 3 === 0 && number % 5 === 0) {
      message = 'FizzBuzz';
    } else if (number % 3 === 0) {
      message = 'Fizz';
    } else if (number % 5 === 0) {
      message = 'Buzz';
    }
    
    console.log(message);
  }
}

fizzBuzz();

// ALTERNATIVE SOLUTION:

// let fizzbuzz = () => {
//   for (let index = 1; index <= 100; index += 1) {
//     let message = index;

//     if (index % 3 === 0 && index % 5 === 0) {
//       message = 'FizzBuzz';
//     } else if (index % 3 === 0) {
//       message = 'Fizz';
//     } else if (index % 5 === 0) {
//       message = 'Buzz';
//     }

//     console.log(message);
//   }
// };

// FURTHER EXPLORATION:

// let fizzbuzz = () => {
//   for (let index = 1; index <= 100; index += 1) {
//     let message = '';

//     if (index % 3 === 0) {
//       message += 'Fizz';
//     }

//     if (index % 5 === 0) {
//       message += 'Buzz';
//     }

//     console.log(message || index);
//   }
// };
